polynote-kernel/src/main/scala/polynote/kernel/util/ZTopic.scala

```diff
@@ -0,0 +1,286 @@
+package polynote.kernel.util
+
+import java.util.concurrent.atomic.AtomicLong
+
+import zio.stream.{Take, ZStream}
+import zio.{Cause, IO, Managed, Promise, Queue, Ref, Semaphore, UIO, UManaged, ZIO, ZQueue}
+
```

Comment:
Not using this yet, but eventually hope to replace fs2 `Topic`s with it. I thought I would need it for `NotebookRef`, but putting the implementations inside the repository implementations made a topic unnecessary.